skimage/measure/tests/test_regionprops.py

```diff
@@ -580,12 +579,22 @@ def test_regionprops_table():
 def test_regionprops_table_equal_to_original():
     regions = regionprops(SAMPLE, INTENSITY_FLOAT_SAMPLE)
     out_table = regionprops_table(SAMPLE, INTENSITY_FLOAT_SAMPLE,
-                                  properties=COL_DTYPES.keys())
-
-    for prop in COL_DTYPES.keys():
-        for i in range(len(regions)):
-            if not isinstance(regions[i][prop], Iterable):
-                assert regions[i][prop] == out_table[prop][i]
+                                  properties=COL_DTYPES.keys(),
+                                  separator="-")
+
+    for prop, dtype in COL_DTYPES.items():
+        for i, reg in enumerate(regions):
+            rp = reg[prop]
+            if np.isscalar(rp) or \
+                    prop in OBJECT_COLUMNS or \
+                    dtype is np.object_:
```

Comment:
```suggestion
            if np.isscalar(rp) or prop in OBJECT_COLUMNS:
```
The `dtype is np.object_` condition is not necessary.

Reply:
Oh thanks, I see! This is actually (dirty) copy from the _regionprops.py logic itself. I'll try to rewrite it in a bit more DRY way.